disable ens if it is not enabled

jrainville opened the pull request:
## Overview
Disabling ENS in config completely stops ens stuff

alaibe commented on lib/modules/ens/index.js:
```diff
@@ -14,6 +14,13 @@ class ENS {
+    if (this.namesConfig === {} ||
+      this.namesConfig.enabled !== true ||
+      this.namesConfig.available_providers.indexOf('ens') < 0 ||
+      this.namesConfig.provider !== 'ens') {
```
what if the provider is `ipns` but `ens` is part of the available providers.
Should it still skip everything?

jrainville replied:
Are the ENS contracts and/or ENS functions needed for IPNS?
Sorry if it's a stupid question, I'm still unsure what is IPNS

iurimatias replied:
IPNS is the equivalent but for IPFS.

@alaibe this is behaviour is correct because this is the ens module, so it only affect the ens stuff.

alaibe replied:
If you have this config
```{enabled: true, available_providers: ['ens', 'ipns'], provider: 'ipns'}```

Both providers are available. According to your conditions, the ens contract will not be deployed because of `this.namesConfig.provider !== 'ens'` but they should

So removing `this.namesConfig.provider !== 'ens'` should make it works
What do you think?

jrainville commented:
I removed `this.namesConfig.provider !== 'ens'`, so that the contracts can deploy if the provider is IPNS, but I added a condition that we don't add ENS to EmbarkJS because it's not the provider. I think that's a correct behaviour.

emizzle commented on lib/modules/ens/index.js:
```diff
@@ -179,12 +187,7 @@ class ENS {
-    }
-
-    if ((this.namesConfig.available_providers.indexOf('ens') < 0) && (this.namesConfig.provider !== 'ens' || this.namesConfig.enabled !== true)) {
+    if (self.providerNotENS) {
```
I think you can remove this because we still want to register the ENS provider, which would allow a Dapp author to do:
`EmbarkJS.Names.setProvider('ens')`
